CME-220: 5000 page size test

AntonyLeons opened the pull request:
### Jira link
https://tools.hmcts.net/jira/browse/CME-220

### Change description

<!--
Provide a description of what change you are proposing.
A short summary here and then you can add comments in your pull request explaining your change to others.
-->

### Testing done

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is not always complete, so a successful pull request build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behaviour remains the same.
-->

### Security Vulnerability Assessment ###

<!-- Comment:
If Yes to the below question, please provide details below:
CVE ID(s): (List all suppressed or relevant CVE IDs)
Reason for Suppression/Ignoring: (e.g., Low risk in our specific context, Mitigating controls in place, False positive - with justification)
Mitigating Factors/Compensating Controls: Describe any measures taken to reduce the risk associated with the vulnerability
-->

**CVE Suppression:** Are there any CVEs present in the codebase (either newly introduced or pre-existing) that are being intentionally suppressed or ignored by this commit?
  * [ ] Yes
  * [ ] No

### Checklist

<!-- Check each box by removing the space and adding an x, e.g. [x] -->

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

AntonyLeons committed:
Refactor user data caching and role assignment logic for improved state management

lint

Update yarn-audit-known-issues

Refactor user data caching and role assignment logic for improved state management

Update yarn.lock

Update yarn.lock

AntonyLeons committed:
Merge branch 'master' into cme-220-2

AntonyLeons committed:
Update yarn.lock

kiran-yenigala-hmcts committed:
Merge branch 'master' into cme-220-2

AntonyLeons committed:
Update yarn-audit-known-issues

kiran-yenigala-hmcts committed:
Merge branch 'master' into cme-220-2

AntonyLeons committed:
attempt test fix

Revert "attempt test fix"

This reverts commit 42e4466172210c3232f30415f676141b469a1ed1.

fix test

Revert "fix test"

This reverts commit 4164d75d651399865eeed958d588c9d9711904b9.

CME-220: fix authtoken issues

Revert "CME-220: fix authtoken issues"

This reverts commit 4d6d0592442a346f8572604e282542cb49cfb7e6.

remove error throwing

connorpgpmcelroy reviewed (dismissed):
LGTM

andywilkinshmcts commented on api/workAllocation/caseWorkerUserDataCacheService.ts:
```diff
@@ -13,81 +13,136 @@ import { CachedCaseworker, LocationApi } from './interfaces/common';
+
+      // Check if this is the last page
+      if (userResponse.length < pageSize) {
+        hasMoreData = false;
```
I think this will go slightly wrong if the last page of results has exactly pageSize results, ie the total number of results is an exact multiple of page size. Probably it will just do a one more request than is needed, for page n+1 of n, but check if this causes an error which could make the whole thing appear to have failed

andywilkinshmcts reviewed (changes requested):
some quite picky comments - sorry. Have a look and see if you think you need to action any of them..

AntonyLeons committed:
fix(caseWorkerUserDataCache): handle missing CASEWORKER_PAGE_SIZE and improve pagination logic

Add fallback value when CASEWORKER_PAGE_SIZE is missing or invalid
Improve pagination termination condition to handle edge cases

AntonyLeons committed:
perf: use concat instead of spread for array merging

Using concat is more efficient than spread operator for merging arrays in this context

AntonyLeons committed:
fix(caseWorkerUserDataCache): reset cache state on critical errors

AntonyLeons committed:
Update yarn-audit-known-issues

connorpgpmcelroy reviewed (dismissed):
LGTM

AntonyLeons committed:
Merge branch 'master' into cme-220-2

AntonyLeons committed:
Update yarn-audit-known-issues

connorpgpmcelroy committed:
Merge branch 'master' into cme-220-2

connorpgpmcelroy committed:
Update package.json

connorpgpmcelroy committed:
Add minor CVE changes

connorpgpmcelroy committed:
Merge branch 'master' into cme-220-2

connorpgpmcelroy committed:
Merge branch 'master' into cme-220-2

connorpgpmcelroy committed:
Merge branch 'master' into cme-220-2

connorpgpmcelroy committed:
Merge branch 'master' into cme-220-2

RiteshHMCTS commented:
Overridden by https://github.com/hmcts/rpx-xui-webapp/pull/5019
